File: problems/problems_2374/Solution.cpp
//go:build ignore
#include "cpp/common/Solution.h"

using namespace std;
using json = nlohmann::json;

class Solution {
public:
  int edgeScore(vector<int> &edges) {
    int ans = 0, n = static_cast<int>(edges.size());
    vector<int64_t> counter(n, 0);
    for (int i = 0; i < n; i++) {
      counter[edges[i]] += i;
      if (counter[edges[i]] > counter[ans]) {
        ans = edges[i];
      } else if (counter[edges[i]] == counter[ans]) {
        ans = min(ans, edges[i]);
      }
    }
    return ans;
  }
};

json leetcode::qubh::Solve(string input_json_values) {
  vector<string> inputArray;
  size_t pos = input_json_values.find('\n');
  while (pos != string::npos) {
    inputArray.push_back(input_json_values.substr(0, pos));
    input_json_values = input_json_values.substr(pos + 1);
    pos = input_json_values.find('\n');
  }
  inputArray.push_back(input_json_values);

  Solution solution;
  vector<int> edges = json::parse(inputArray.at(0));
  return solution.edgeScore(edges);
}

File: problems/problems_2374/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn edge_score(edges: Vec<i32>) -> i32 {
		let mut ans: i32 = 0;
		let n = edges.len();
		let mut counter: Vec<i64> = vec![0; n];
		for i in 0..n {
			counter[edges[i] as usize] += i as i64;
			if counter[edges[i] as usize] > counter[ans as usize] {
				ans = edges[i];
			} else if counter[edges[i] as usize] == counter[ans as usize] {
				ans = ans.min(edges[i]);
			}
		}
		ans
    }
}

#[cfg(feature = "solution_2374")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let edges: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	json!(Solution::edge_score(edges))
}
